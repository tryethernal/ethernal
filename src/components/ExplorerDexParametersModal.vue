<template>
    <v-dialog v-model="dialog" max-width="350">
        <v-card outlined>
            <v-card-title>
                Transaction Settings
                <v-spacer></v-spacer>
                <v-btn icon @click="close()" ><v-icon>mdi-close</v-icon></v-btn>
            </v-card-title>
            <v-card-text>
                Slippage tolerance
                <div class="d-flex">
                    <v-chip-group mandatory active-class="primary--text text--accent-4" v-model="slippageTolerance">
                        <v-chip small class="mr-2" :value="0.1">0.1%</v-chip>
                        <v-chip small class="mr-2" :value="0.5">0.5%</v-chip>
                        <v-chip small class="mr-2" :value="1">1%</v-chip>
                    </v-chip-group>
                    <v-text-field
                        dense
                        reverse
                        type="number"
                        class="rounded-xl ml-5"
                        prefix="%"
                        hide-details="auto"
                        outlined
                        v-model="slippageTolerance">
                    </v-text-field>
                </div>
                <v-divider class="my-3"></v-divider>
                Transaction deadline
                <v-text-field
                    style="max-width: 140px;"
                    dense
                    reverse
                    type="number"
                    class="rounded-xl mt-1"
                    prefix="minutes"
                    hide-details="auto"
                    outlined
                    v-model="transactionTimeoutInMinutes"></v-text-field>
            </v-card-text>
        </v-card>
    </v-dialog>
</template>
<script>

export default {
    name: 'ExplorerDexParametersModal',
    data: () => ({
        dialog: false,
        resolve: null,
        reject: null,
        transactionTimeout: null,
        slippageToleranceInBps: null,
        options: null
    }),
    methods: {
        open(options) {
            this.dialog = true;
            this.options = options;
            this.transactionTimeout = options.transactionTimeout;
            this.slippageToleranceInBps = options.slippageToleranceInBps;
            return new Promise((resolve, reject) => {
                this.resolve = resolve;
                this.reject = reject;
            });
        },
        update() {
            this.$emit('parametersChanged', {
                transactionTimeout: this.transactionTimeout,
                slippageToleranceInBps: this.slippageToleranceInBps
            });
        },
        close() {
            this.resolve();
            this.options = null;
            this.dialog = false;
        }
    },
    computed: {
        slippageTolerance: {
            get() {
                if (!this.slippageToleranceInBps)
                    return '';
                return parseFloat(this.slippageToleranceInBps) / 100;
            },
            set(val) {
                this.slippageToleranceInBps = parseFloat(val) * 100;
                this.update();
            }
        },
        transactionTimeoutInMinutes: {
            get() {
                if (!this.transactionTimeout)
                    return 0;
                return Math.floor(this.transactionTimeout / 60);
            },
            set(val) {
                this.transactionTimeout = val * 60;
                this.update();
            }
        }
    }
}
</script>
<style scoped>
.v-text-field__slot input {
    text-align: right;
}
</style>
