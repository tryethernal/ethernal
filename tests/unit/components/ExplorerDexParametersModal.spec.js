import flushPromises from 'flush-promises';
import MockHelper from '../MockHelper';

import ExplorerDexParametersModal from '@/components/ExplorerDexParametersModal.vue';

let helper;
beforeEach(() => {
    jest.clearAllMocks()
    helper = new MockHelper();
});

describe('ExplorerDexParametersModal.vue', () => {
    it('Should display dex parameters', async () => {
        const wrapper = helper.mountFn(ExplorerDexParametersModal, {
            data() {
                return {
                    dialog: true,
                    resolve: jest.fn().mockResolvedValue(),
                    transactionTimeout: 20 * 60 * 60,
                    slippageToleranceInBps: 0.5 * 100
                };
            }
        });
        await flushPromises();

        expect(wrapper.html()).toMatchSnapshot();
    });
});
