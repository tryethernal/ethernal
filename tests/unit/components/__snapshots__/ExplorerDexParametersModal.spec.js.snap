// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExplorerDexParametersModal.vue Should display dex parameters 1`] = `
<div class="v-dialog__container">
  <div role="dialog" aria-modal="true" class="v-dialog__content v-dialog__content--active" style="z-index: 202;">
    <transition-stub name="dialog-transition" appear="true">
      <div tabindex="0" class="v-dialog v-dialog--active" style="transform-origin: center center; max-width: 350px;">
        <div class="v-card v-sheet v-sheet--outlined theme--light">
          <div class="v-card__title">
            Transaction Settings
            <div class="spacer"></div> <button type="button" class="v-btn v-btn--icon v-btn--round theme--light v-size--default"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-close theme--light"></i></span></button>
          </div>
          <div class="v-card__text">
            Slippage tolerance
            <div class="d-flex">
              <div class="v-item-group theme--light v-slide-group v-chip-group">
                <div class="v-slide-group__prev v-slide-group__prev--disabled">
                  <transition-stub name="fade-transition"></transition-stub>
                </div>
                <div class="v-slide-group__wrapper">
                  <div class="v-slide-group__content"><span tabindex="0" class="mr-2 v-chip v-chip--clickable v-chip--no-color theme--light v-size--small"><span class="v-chip__content">0.1%</span></span> <span tabindex="0" class="mr-2 v-chip primary--text text--accent-4 v-chip--active v-chip--clickable v-chip--no-color theme--light v-size--small"><span class="v-chip__content">0.5%</span></span> <span tabindex="0" class="mr-2 v-chip v-chip--clickable v-chip--no-color theme--light v-size--small"><span class="v-chip__content">1%</span></span></div>
                </div>
                <div class="v-slide-group__next v-slide-group__next--disabled">
                  <transition-stub name="fade-transition"></transition-stub>
                </div>
              </div>
              <div class="v-input rounded-xl ml-5 v-input--hide-details v-input--is-label-active v-input--is-dirty v-input--dense theme--light v-text-field v-text-field--prefix v-text-field--enclosed v-text-field--reverse v-text-field--outlined">
                <div class="v-input__control">
                  <div class="v-input__slot">
                    <fieldset aria-hidden="true">
                      <legend style="width: 0px;"><span class="notranslate">​</span></legend>
                    </fieldset>
                    <div class="v-text-field__slot">
                      <div class="v-text-field__prefix">%</div><input id="input-16" type="number">
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <hr role="separator" aria-orientation="horizontal" class="my-3 v-divider theme--light">
            Transaction deadline
            <div class="v-input rounded-xl mt-1 v-input--hide-details v-input--is-label-active v-input--is-dirty v-input--dense theme--light v-text-field v-text-field--prefix v-text-field--enclosed v-text-field--reverse v-text-field--outlined" style="max-width: 140px;">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <fieldset aria-hidden="true">
                    <legend style="width: 0px;"><span class="notranslate">​</span></legend>
                  </fieldset>
                  <div class="v-text-field__slot">
                    <div class="v-text-field__prefix">minutes</div><input id="input-18" type="number">
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </transition-stub>
  </div>
</div>
`;
